packages/twenty-server/src/engine/workspace-manager/dev-seeder/data/services/timeline-activity-seeder.service.ts

```diff
@@ -0,0 +1,395 @@
+import { Injectable } from '@nestjs/common';
+
+import chunk from 'lodash.chunk';
+
+import { ObjectMetadataService } from 'src/engine/metadata-modules/object-metadata/object-metadata.service';
+import { WorkspaceEntityManager } from 'src/engine/twenty-orm/entity-manager/workspace-entity-manager';
+import { COMPANY_DATA_SEEDS } from 'src/engine/workspace-manager/dev-seeder/data/constants/company-data-seeds.constant';
+import { NOTE_DATA_SEEDS } from 'src/engine/workspace-manager/dev-seeder/data/constants/note-data-seeds.constant';
+import { NOTE_TARGET_DATA_SEEDS } from 'src/engine/workspace-manager/dev-seeder/data/constants/note-target-data-seeds.constant';
+import { OPPORTUNITY_DATA_SEEDS } from 'src/engine/workspace-manager/dev-seeder/data/constants/opportunity-data-seeds.constant';
+import { PERSON_DATA_SEEDS } from 'src/engine/workspace-manager/dev-seeder/data/constants/person-data-seeds.constant';
+import { TASK_DATA_SEEDS } from 'src/engine/workspace-manager/dev-seeder/data/constants/task-data-seeds.constant';
+import { TASK_TARGET_DATA_SEEDS } from 'src/engine/workspace-manager/dev-seeder/data/constants/task-target-data-seeds.constant';
+import { WORKSPACE_MEMBER_DATA_SEED_IDS } from 'src/engine/workspace-manager/dev-seeder/data/constants/workspace-member-data-seeds.constant';
+
+@Injectable()
+export class TimelineActivitySeederService {
+  constructor(private readonly objectMetadataService: ObjectMetadataService) {}
+
+  async seedTimelineActivities({
+    entityManager,
+    schemaName,
+    workspaceId,
+  }: {
+    entityManager: WorkspaceEntityManager;
+    schemaName: string;
+    workspaceId: string;
+  }) {
+    const timelineActivities: Record<string, unknown>[] = [];
+
+    const { noteMetadataId, taskMetadataId } =
+      await this.getObjectMetadataIds(workspaceId);
+
+    let activityIndex = 0;
+
+    const entityConfigs = [
+      { type: 'company', seeds: COMPANY_DATA_SEEDS },
+      { type: 'person', seeds: PERSON_DATA_SEEDS },
+      { type: 'note', seeds: NOTE_DATA_SEEDS },
+      { type: 'task', seeds: TASK_DATA_SEEDS },
+      { type: 'opportunity', seeds: OPPORTUNITY_DATA_SEEDS },
+    ];
+
+    entityConfigs.forEach(({ type, seeds }) => {
```

Comment:
for (entityConfig of entityConfigs) {

// you can also extract the index from for ... of syntax